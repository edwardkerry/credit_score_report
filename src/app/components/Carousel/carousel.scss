.carousel__container {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  ul {
    padding: 0;
    margin: 0;
    list-style-type: none;
  }
}

.carousel__dots {
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin-top: 20px;
  li {
    &:nth-of-type(n + 2) {
      margin-left: 9px;
    }
  }
}