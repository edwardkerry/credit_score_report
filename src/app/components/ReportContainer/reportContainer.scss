.report__container{
  display: flex;
  margin-top: 20vh;
  margin-bottom: 25vh;
  border: solid rgba(255, 255, 255, 0.932) 2px;
  border-radius: 50%;
  background: rgba(110, 107, 107, 0.5);
  height: 400px;
  width: 400px;
  overflow: hidden;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 25px;
  color: white;
  text-align: center;
  
  @media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
    font-size: 15px;
    height: 250px;
    width: 250px;
  }

  .slider-list {
    margin-top: 75px !important;
    @media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
      margin-top: 50px !important;
    }
  }
}