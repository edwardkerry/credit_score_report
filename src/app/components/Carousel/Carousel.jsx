import React, { Component } from 'react';
import { array } from 'prop-types';
import bemHelper from '../../utils/bem';
import Slide from './Slide/Slide.jsx';
import Dot from './Dot/Dot.jsx';
import './carousel.scss';
import { removeDotSegments } from 'uri-js';

const cn = bemHelper({ block: 'carousel' });

class Carousel extends Component {

  constructor(props) {
    super(props);
    this.state = {
      activeSlideIndex: 0
    };

    this.keyCount = 0;
    this.getKey = this.getKey.bind(this);
    this.selectSlide = this.selectSlide.bind(this)
  };
  
  selectSlide(index) {
    this.setState({ activeSlideIndex: index });
  };

  getKey() {
    return `CarouselKey${this.keyCount ++}`
  }

  render() {
    return (
      <div className={cn('container', null)}>
        <ul className={cn('slides', null)}>
          {this.props.slides.map((slide, index) =>{
           return(<Slide 
              content={slide}
              active={index == this.state.activeSlideIndex}
              key={this.getKey()}
            />)
          })}
        </ul>

        <ul className={cn('dots', null)}>
          {this.props.slides.map((slide, index) => {
            return(<Dot
              select={() => this.selectSlide(index)}
              active={index == this.state.activeSlideIndex}
              key={this.getKey()}
            />) 
          })}
        </ul>
      </div>
    );
  };
};

Carousel.propTypes = {
  slides: array
};

export default Carousel;