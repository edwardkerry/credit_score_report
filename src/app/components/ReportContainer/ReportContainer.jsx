import React, { Component } from 'react';
import {shape, number, string } from 'prop-types';

import Carousel from 'nuka-carousel';
import CreditScoreReport from '../CreditScoreReport/CreditScoreReport.jsx';
import LongTermDebtReport from '../LongTermDebtReport/LongTermDebtReport.jsx';

import bemHelper from '../../utils/bem';

const cn = bemHelper({ block: 'report' });

const ReactContainer = ({ creditScore, longTermDebt}) => {

  const { score, maxScore, description } = { ...creditScore };
  const { currentCreditLimit, currentDebt, debtChange} = { ...longTermDebt }

   return (
    <Carousel className={cn('carousel', null)}>
      <CreditScoreReport 
        score={score} 
        maxScore={maxScore} 
        description={description} 
      />
      <LongTermDebtReport 
        currentCreditLimit={currentCreditLimit} 
        currentDebt={currentDebt} 
        debtChange={debtChange}
      />
    </Carousel>
  );
};

ReactContainer.propTypes = {
  creditScore: shape({
    description: string.isRequired,
    maxScore: number.isRequired,
    score: number.isRequired 
  }),

  longTermDebt:  shape({
    currentDebt: number.isRequired,
    currentCreditLimit: number.isRequired,
    debtChange: number.isRequired
  })
}

export default ReactContainer;
