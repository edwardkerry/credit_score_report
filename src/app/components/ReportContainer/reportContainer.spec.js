import React from 'react';
import renderer from 'react-test-renderer';
import Carousel from 'nuka-carousel';

import { creditScoreReport, longTermDebtReport} from '../../../../tests/mocks/fixtures'

import ReportContainer from '../ReportContainer/ReportContainer.jsx'

describe('ReportContainer', () => {
  
  it('renders', () => {
    const component = renderer.create(<ReportContainer creditScore={creditScoreReport} longTermDebt={longTermDebtReport} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

});