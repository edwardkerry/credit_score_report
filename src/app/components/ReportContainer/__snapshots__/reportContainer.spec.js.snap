// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactContainer renders 1`] = `
<div
  className="report__loading"
/>
`;
